### PR TITLE
Jh/form wizard

### LINKED ISSUE #1192: Create a Wizard to get to know visitors to dev.near.org
Let's add into dev.near.org a wizard that asks users questions which would help us better understand our developers needs.  

[UX Designs](https://www.figma.com/design/LeXoElysTj0ckjB4vKZti5/Dev.near.org-In-Product-Continuous-Discovery-Wizard?node-id=0-1&t=4U2QMlSco7pxsGVb-0)
[Mobile UX](https://www.figma.com/design/LeXoElysTj0ckjB4vKZti5/Dev.near.org-In-Product-Continuous-Discovery-Wizard?node-id=60-1879&t=OkeDLrMXOKWvo6Jw-0)

## Additional Requirements

1. The wizard [should not appear](https://www.figma.com/design/LeXoElysTj0ckjB4vKZti5?node-id=48-932#860605361) for users who only visit `/applications`. 
2. The wizard should only appear after the Cookie Consent pop-up has been acknowledged.
3. Similar to our cookie consent pop-up, this wizard should only appear once per user session
4. One each click of 'next' we can [send the data to rudderstack with a specific event payload](https://github.com/near/near-discovery/blob/d9e9cea72a87ca13824462b468db3b38e2cbcbe8/src/utils/analytics.ts#L186) such as 
` "event": "q3_developer_survey",
    "properties": { 
        "question": 
        "answer":
         ....
     } 
`
5. @jackson-harris-iii also send an event to track the display of the wizard. So we know a total number of times the wizard was displayed so we can compare that to the number of people who actually interact with the wizard. 

Coinbase does a nice job with this. 
coinbase

### PR DESCRIPTION
**Pull request by jackson-harris-iii**
This PR adds a "get to know you form on the dev.near.org website. 

- The implementation uses local storage for completion/opt out tracking
- Rudderstack is usesd for capturing user feedback and emails.
- The implementation is mobile optimized and is based off of this [Figma](https://www.figma.com/design/LeXoElysTj0ckjB4vKZti5/Dev.near.org-Research-Recruitment-Wizard?node-id=0-1&t=4XqfCtBrSOq59kcH-1).

Here is an example of the payloads sent to rudderstack. :

`...
    "event": "q3-developer-survey-submission",
    "properties": {
        "questionNumber": "1",
        "response": {
            "interests": [
                "Docs",
                "Evaluating tech",
                "Starting a new project"
            ],
            "other": ""
        },
      ...
    "sentAt": "2024-07-16T22:22:15.093Z"
}`
 
Closes #1192

### CODE REVIEW

**Comment by charleslavon** [comment]
@jackson-harris-iii I really appreciate your well-organized code, components, and your semantic approach to layering UI elements 👏🏾

**Comment by charleslavon** [comment]
hey @xuan-ytr @vanessadespain when you have time, please take a look at [this preview](https://vercel.live/open-feedback/dev-near-org-git-jh-formwizard-near-developer-console.vercel.app?via=pr-comment-visit-preview-link&passThrough=1) and share any feedback on the UI.  The wizard will appear shortly after accepting T&Cs and connecting a wallet.

File: src/components/research-form-wizard/FormPages/FollowUp.tsx
Comment (shelegdmitriy) [inline src/components/research-form-wizard/FormPages/FollowUp.tsx:191]: Maybe we should add links to terms and privacy pages here?

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
Знімок екрана 2024-07-15 о 15 31 21

Seems like some styles are missing on the desktop

File: src/components/research-form-wizard/ResearchFormWizard.tsx
Comment (charleslavon) [inline src/components/research-form-wizard/ResearchFormWizard.tsx:88]: @jackson-harris-iii Let's use `q3_developer_survey` instead of the prefix `research-form`

File: src/components/research-form-wizard/ResearchFormWizard.tsx
Comment (charleslavon) [inline src/components/research-form-wizard/ResearchFormWizard.tsx:78]: @jackson-harris-iii Did Jo or Spencer recommend this structure? Recording an index would require a look-up against some other data store to resolve what the action question was.  Why not require the full info such as `"event": "q3_developer_survey", "properties": {  "question": "Which of the following best describes you..."  "answer": "Other - wen 🚀 ?"}`

Comment (charleslavon) [reply src/components/research-form-wizard/ResearchFormWizard.tsx:78]: Or perhaps we are recording the full question somewhere and I have missed it.

File: src/components/research-form-wizard/FormPages/FollowUp.tsx
Comment (charleslavon) [reply src/components/research-form-wizard/FormPages/FollowUp.tsx:191]: @jackson-harris-iii  https://dev.near.org/terms & https://dev.near.org/privacy

File: src/components/research-form-wizard/StepLayout.tsx
Comment (charleslavon) [inline src/components/research-form-wizard/StepLayout.tsx:135]: Let's use the ph icon set here: `ph ph-x`

File: src/utils/analytics.ts
Comment (charleslavon) [inline src/utils/analytics.ts:206]: @jackson-harris-iii Use `recordEventWithProps` directly within `ResearchFormWizard`, since there's no need to expose this method to other consumers of `analytics.ts`

File: src/components/research-form-wizard/ResearchFormWizard.tsx
Comment (jackson-harris-iii) [reply src/components/research-form-wizard/ResearchFormWizard.tsx:78]: The dismiss is when the user clicks the x, this allows us to record that a user is dismissing the form.  We record interactions with the question information and their answers in 'handleFormButton', this function fires off the submission on each click of the next or submit button :

`const handleFormButton = () => {
    if (currentStepSubmission) {
      recordResearchFromEvent('q3-developer-survey-submission', {
        questionNumber: currentStepIndex + 1,
        response: currentStepSubmission,
      });

      if (currentStepIndex < formSteps.length - 1) {
        setCurrentStep(formSteps[currentStepIndex + 1].component());
        useResearchWizardStore.setState({ currentStepIndex: currentStepIndex + 1 });
      } else {
        setIsResearchFormDismissed(true);
        localStorage.setItem('researchFormDismissed', 'true');
      }
      set({ currentStepSubmission: '' });
    }
  };`

Comment (charleslavon) [reply src/components/research-form-wizard/ResearchFormWizard.tsx:78]: https://github.com/near/near-discovery/pull/1251/files#diff-48f525b8d86b2469507944329205ec412edc43cc9a3e5f4ac11af7408d153eebR89

**Comment by jackson-harris-iii** [comment]
@race-of-sloths please, include my PR in the Race?

**Comment by race-of-sloths** [comment]
### 🙁 This repository is not a part of the [Race of Sloths](https://race-of-sloths.com)
Please ask the maintainer to [contact us](https://github.com/NEAR-DevHub/race-of-sloths/issues/new/choose) to join the race!

<details><summary><strong>What is the Race of Sloths</strong></summary>
<p>

Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag us inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/jackson-harris-iii)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!

</p>
</details>

**Comment by shelegdmitriy** [comment]
Nice work! One thing I'd like to add here is success toast message after submitting form. Along with this, let's add a error prompt if the request failed

File: src/components/research-form-wizard/FormPages/FollowUp.tsx
Comment (shelegdmitriy) [inline src/components/research-form-wizard/FormPages/FollowUp.tsx:83]: Let's use `$` to avoid adding this properties to DOM
```suggestion
  display: ${(props) => (props.$visible ? 'block' : 'none')};
```

File: src/components/research-form-wizard/ResearchFormWizard.tsx
Comment (shelegdmitriy) [inline src/components/research-form-wizard/ResearchFormWizard.tsx:145]: I think we can use ph-icon here https://phosphoricons.com/?q=%22thumb%22

**Review by shelegdmitriy — APPROVED** [review]

**Comment by shelegdmitriy** [comment]
I added a few comments but they are not required so feel free to skip them 🙂

**Pull request merged**
